change: allow full K3s registry configuration

maxsargentdev opened the pull request:
What

Allow rewrites configuration via config file or inline file to be reflected in generated registries.yaml file via merge.

# Why

Provide rewrite functionality from k3s private registries

# Implications

No impact for previous versions.

iwilltry42 commented:
Hi @maxsargentdev thank you for this PR!
I just started working on this when I saw this incoming.
Can you try to replace the custom Registry implementation directly with https://github.com/rancher/wharfie/tree/main/pkg/registries ?
It should be fairly straightforward and we'd get rid of the registry.go file.

maxsargentdev commented:
Sure I will give it a go after work today :)

iwilltry42 commented:
Hey @maxsargentdev did you have any luck yet? :)

maxsargentdev commented:
@iwilltry42 is this still needing doing? Got some free time this week.

maxsargentdev commented:
Roger that will take a look over the next week.

maxsargentdev commented:
@iwilltry42 

So I have dropped the pkg/types/k3s/registry.go containing the registry types and replaced it with a wharfie import. I have updated other references and unit tests are passing, which I would expect as not much has really changed.

Is this the change you had in mind? I have left my original change to pkg/client/registry.go to allow for rewrites.

iwilltry42 commented:
@maxsargentdev yep, that's it!
Thank you very much for this! :pray:
